hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/Iterators.scala

```diff
@@ -302,33 +300,49 @@ class RecordMergingFileIterator(logFiles: List[HoodieLogFile],
   private def merge(curRow: InternalRow, newRecord: HoodieRecord[_]): Option[InternalRow] = {
     // NOTE: We have to pass in Avro Schema used to read from Delta Log file since we invoke combining API
     //       on the record from the Delta Log
-    recordMerger.getRecordType match {
-      case HoodieRecordType.SPARK =>
-        val curRecord = new HoodieSparkRecord(curRow, readerSchema)
-        val result = recordMerger.merge(curRecord, baseFileReaderAvroSchema, newRecord, logFileReaderAvroSchema, payloadProps)
-        toScalaOption(result)
-          .flatMap { r =>
-            val data = r.getLeft.getData.asInstanceOf[InternalRow]
-            if (isDeleteOperation(data)) {
-              None
-            } else {
-              val schema = HoodieInternalRowUtils.getCachedSchema(r.getRight)
-              val projection = HoodieInternalRowUtils.getCachedUnsafeProjection(schema, structTypeSchema)
-              Some(projection.apply(data))
+
+    // Delete records are in-between; no merge is needed.
+    if (newRecord.getMetaDataInfo(HoodieReaderContext.INTERNAL_META_OPERATION).isPresent) {
```

Comment:
lets add an example and say why we need this processing.

Reply:
Will do.